common/scripted_effects/wc_history_effects.txt

```diff
@@ -21,4 +21,242 @@ add_patron_trait_character_effect = {
+		limit = { has_variable = wc_$MAGIC$_magic_lifestyle_additional_perks_variable }
+
+		if = {
+			limit = {
+				has_character_flag = wc_$MAGIC$_magic_lifestyle_perks_were_reset_flag
+			}
+
+			add_$MAGIC$_magic_lifestyle_perk_points = var:wc_$MAGIC$_magic_lifestyle_additional_perks_variable
+		}
+		else = {
+			while = {
+				count = var:wc_$MAGIC$_magic_lifestyle_additional_perks_variable
+
+				random_list = {
+					1 = {
+						trigger = {
+							perks_in_tree = { tree = $MAGIC$_magic_tree_1 value < 9 }
+						}
+
+						add_1_perk_in_TYPE_TREE_effect = { TYPE = $MAGIC$_magic TREE = 1 }
+					}
+					1 = {
+						trigger = {
+							perks_in_tree = { tree = $MAGIC$_magic_tree_2 value < 9 }
+						}
+
+						add_1_perk_in_TYPE_TREE_effect = { TYPE = $MAGIC$_magic TREE = 2 }
+					}
+					1 = {
+						trigger = {
+							perks_in_tree = { tree = $MAGIC$_magic_tree_3 value < 9 }
+						}
+
+						add_1_perk_in_TYPE_TREE_effect = { TYPE = $MAGIC$_magic TREE = 3 }
+					}
+				}
+			}	
+		}
+	}
+}
+
+add_perks_in_elemental_MAGIC_lifestyle_effect = {
+	if = {
+		limit = { has_variable = wc_$MAGIC$_magic_lifestyle_additional_perks_variable }
+
+		if = {
+			limit = {
+				has_character_flag = wc_$MAGIC$_magic_lifestyle_perks_were_reset_flag
+			}
+
+			add_$MAGIC$_magic_lifestyle_perk_points = var:wc_$MAGIC$_magic_lifestyle_additional_perks_variable
+		}
+		else = {
+			while = {
+				count = var:wc_$MAGIC$_magic_lifestyle_additional_perks_variable
+
+				random_list = {
+					1 = {
+						trigger = {
+							perks_in_tree = { tree = $MAGIC$_magic_tree_1 value < 9 }
+						}
+
+						add_1_perk_in_TYPE_TREE_effect = { TYPE = $MAGIC$_magic TREE = 1 }
+					}
+					1 = {
+						trigger = {
+							perks_in_tree = { tree = $MAGIC$_magic_tree_2 value < 9 }
+						}
+
+						add_1_perk_in_TYPE_TREE_effect = { TYPE = $MAGIC$_magic TREE = 2 }
+					}
+				}
+			}	
+		}
+	}
+}
+
+add_perks_in_physical_lifestyle_effect = {
+	if = {
+		limit = {
+			OR = {
+				has_variable = wc_warrior_lifestyle_additional_perks_variable
+				has_variable = wc_rogue_lifestyle_additional_perks_variable
+				has_variable = wc_hunter_lifestyle_additional_perks_variable
+			}
+		}
+
+		set_variable = { name = wc_physical_lifestyle_additional_perks_variable value = 0 days = 10 }
+		if = {
+			limit = { has_variable = wc_warrior_lifestyle_additional_perks_variable }
+			change_variable = { name = wc_physical_lifestyle_additional_perks_variable add = var:wc_warrior_lifestyle_additional_perks_variable }
+		}
+		if = {
+			limit = { has_variable = wc_rogue_lifestyle_additional_perks_variable }
+			change_variable = { name = wc_physical_lifestyle_additional_perks_variable add = var:wc_rogue_lifestyle_additional_perks_variable }
+		}
+		if = {
+			limit = { has_variable = wc_hunter_lifestyle_additional_perks_variable }
+			change_variable = { name = wc_physical_lifestyle_additional_perks_variable add = var:wc_hunter_lifestyle_additional_perks_variable }
+		}
+
+		# while = {
```

Comment:
It's commented.

Vek'nilash is a shadow priest though supposed to be a warrior.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/138368400-78e7b372-6580-4da1-b042-47b57e20e6e6.png)
![image](https://user-images.githubusercontent.com/46384992/138368426-ca369b32-bf77-430b-b3b8-034428e296d5.png)

</details>

Reply:
> It's commented.

Added physical perks

> Vek'nilash is a shadow priest though supposed to be a warrior.

It should be fixed on the later branch that fixes auto selection weight. I can't fix it on this PR. So, please, only rate giving perks to characters from history by event and not from automatic perk allocation by the game.